nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateJson.java

```diff
@@ -0,0 +1,215 @@
+
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        final AtomicReference<Exception> exceptions = new AtomicReference<>();
+        final AtomicReference<Set<ValidationMessage>> validationErrors = new AtomicReference<Set<ValidationMessage>>(null);
+
+        session.read(flowFile, new InputStreamCallback()  {
+
+            @Override
+            public void process(InputStream in) {
+                try {
+                    // Set JSON schema version to use from processor property
+                    VersionFlag schemaVersion = VersionFlag.V201909;
+                    if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_4.getValue()) {
+                        schemaVersion = VersionFlag.V4;
+                    }
+                    if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_6.getValue()) {
+                        schemaVersion = VersionFlag.V6;
+                    }
+                    if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_7.getValue()) {
+                        schemaVersion = VersionFlag.V7;
+                    }
+                    if (context.getProperty(SCHEMA_VERSION).getValue() == SCHEMA_VERSION_V201909.getValue()) {
+                        schemaVersion = VersionFlag.V201909;
+                    }
+        
+                    // Read in flowFile inputstream, and validate against schema
+                    ObjectMapper mapper = new ObjectMapper();
+                    JsonNode node = mapper.readTree(in);
+                    String schemaText = context.getProperty(SCHEMA_TEXT).evaluateAttributeExpressions().getValue();
+                    JsonSchemaFactory factory = JsonSchemaFactory.getInstance(schemaVersion);
+                    JsonSchema schema = factory.getSchema(schemaText);
+                    validationErrors.set(schema.validate(node));
+                
+                } catch (JsonParseException jpe) {
+                    exceptions.set(jpe);
+                } catch (IOException ioe) {
+                    exceptions.set(ioe);
+                }
+            }
+        });
+
+        // Failed to read flowFile - either IOException, or JsonParseException
+        if (exceptions.get() != null) {
+            this.getLogger().info("Failed to process {} due to {}; routing to 'failure'", new Object[]{flowFile, exceptions.get().getLocalizedMessage()});
+            session.getProvenanceReporter().route(flowFile, REL_FAILURE);
```

Comment:
We should not create a `ROUTE` event for failure. It is a common flow design pattern to loop failure back to itself. Doing this could result in creating millions of Provenance Events just indicating that the data kept looping.